airflow/www/utils.py

```diff
@@ -779,6 +787,72 @@ class UtcAwareFilterConverter(fab_sqlafilters.SQLAFilterConverter):
     """Retrieve conversion tables for UTC-Aware filters."""
 
 
+class XComFilterStartsWith(fab_sqlafilters.FilterStartsWith):
+    """Starts With filter for XCom values."""
+
+    def apply(self, query: Query, value: str) -> Query:
+        query, trimmed_value = get_trimmed_field(query, self.model, self.column_name)
+        return query.filter(trimmed_value.ilike(value + "%"))
+
+
+class XComFilterEndsWith(fab_sqlafilters.FilterEndsWith):
+    """Ends With filter for XCom values."""
+
+    def apply(self, query: Query, value: str) -> Query:
+        query, trimmed_value = get_trimmed_field(query, self.model, self.column_name)
+        return query.filter(trimmed_value.ilike("%" + value))
+
+
+class XComFilterEqual(fab_sqlafilters.FilterEqual):
+    """Equality filter for XCom values."""
+
+    def apply(self, query: Query, value: str) -> Query:
+        query, trimmed_value = get_trimmed_field(query, self.model, self.column_name)
+        value = set_value_to_type(self.datamodel, self.column_name, value)
+        return query.filter(trimmed_value == value)
+
+
+class XComFilterContains(fab_sqlafilters.FilterContains):
+    """Not Equal To filter for XCom values."""
+
+    def apply(self, query: Query, value: str) -> Query:
+        query, trimmed_value = get_trimmed_field(query, self.model, self.column_name)
+        return query.filter(trimmed_value.ilike("%" + value + "%"))
+
+
+class XComFilterNotStartsWith(fab_sqlafilters.FilterNotStartsWith):
+    """Not Starts With filter for XCom values."""
+
+    def apply(self, query: Query, value: str) -> Query:
+        query, trimmed_value = get_trimmed_field(query, self.model, self.column_name)
+        return query.filter(~trimmed_value.ilike(value + "%"))
+
+
+class XComFilterNotEndsWith(fab_sqlafilters.FilterNotEndsWith):
+    """Not Starts With filter for XCom values."""
+
+    def apply(self, query: Query, value: str) -> Query:
+        query, trimmed_value = get_trimmed_field(query, self.model, self.column_name)
+        return query.filter(~trimmed_value.ilike("%" + value))
+
+
+class XComFilterNotContains(fab_sqlafilters.FilterNotContains):
+    """Not Starts With filter for XCom values."""
+
+    def apply(self, query: Query, value: str) -> Query:
+        query, trimmed_value = get_trimmed_field(query, self.model, self.column_name)
+        return query.filter(~trimmed_value.ilike("%" + value + "%"))
+
+
+class XComFilterNotEqual(fab_sqlafilters.FilterNotEqual):
+    """Not Starts With filter for XCom values."""
+
+    def apply(self, query: Query, value: str) -> Query:
+        query, trimmed_value = get_trimmed_field(query, self.model, self.column_name)
+        value = set_value_to_type(self.datamodel, self.column_name, value)
+        return query.filter(trimmed_value != value)
```

Comment:
follow up of the "another idea" above

```suggestion
   def apply(self, query: Query, value: str) -> Query:
        value = set_value_to_type(self.datamodel, self.column_name, value)
        return generate_filter_value_query(
            query=query,
            model=self.model,
            column_name=self.column_name, 
            filter_cond=lambda trimmed_value: trimmed_value != value
        )
```